### PR TITLE
A flow for communication between the plugin builder and the publishers

### LINKED ISSUE #40: Provide a button to email all plugin authors from Administration
We now have an EmailSender page that accepts parameters to initialize To, Subject, and Message fields; we can leverage this to message all the plugin authors. Initially, we will invite them to verify their email accounts.

**What's needed:**
- Add an "Email all Publishers" button on the `admin/plugins` page.
- Clicking this button will redirect to the `admin/emailsender` page and populate the `To` textarea with a comma-separated list of all the authors' emails.
- Validate the `To` field to ensure it contains valid emails, adhering to formats such as `hello@world.com` or `Hello World <hello@world.com>`. The system should ensure that a comma-separated list of valid emails passes validation but throws an error if it does not.
- Leave the `Subject` and `Email Message` fields blank and ensure all required field validations are functioning properly.
- On the server side, ensure that validations are thorough and that emails are sent independently to maintain privacy (preventing one recipient from seeing another’s address).
- Display a success message indicating the number of emails sent, while preserving all sending parameters (To, Subject, Message).

Here is a video showing the areas in question:

https://github.com/user-attachments/assets/a50f3147-2ef3-4fe4-af4c-2c13179e5aac

### PR DESCRIPTION
**Pull request by TChukwuleta**
Resolves #40

### CODE REVIEW

File: PluginBuilder/Controllers/AdminController.cs [hunk 1]
```diff
@@ -312,18 +313,21 @@ private async Task SaveEmailSettingsToDatabase(EmailSettingsViewModel model)
...
-    //
 
     [HttpGet("emailsender")]
     public async Task<IActionResult> EmailSender(string to, string subject, string message)
     {
+        if (!string.IsNullOrEmpty(to) && !IsValidEmailList(to))
+        {
+            TempData[TempDataConstant.WarningMessage] = "Invalid email format in the 'To' field. Please ensure all emails are valid";
+            return RedirectToAction("Plugins");
+        }
```
Comment (rockstardev) [inline PluginBuilder/Controllers/AdminController.cs:324]: This needs to be ViewModel error that is assigned to field

File: PluginBuilder/Controllers/AdminController.cs [hunk 2]
```diff
@@ -312,18 +313,21 @@ private async Task SaveEmailSettingsToDatabase(EmailSettingsViewModel model)
...
     }
-    //
 
     [HttpGet("emailsender")]
     public async Task<IActionResult> EmailSender(string to, string subject, string message)
     {
+        if (!string.IsNullOrEmpty(to) && !IsValidEmailList(to))
+        {
+            TempData[TempDataConstant.WarningMessage] = "Invalid email format in the 'To' field. Please ensure all emails are valid";
+            return RedirectToAction("Plugins");
```
Comment (rockstardev) [inline PluginBuilder/Controllers/AdminController.cs:323]: This is an invalid redirection... and we also shouldn't redirect to Plugins; rather we should provide in `ModelState.AddModelError` for To field that invalid emails were provided.

File: PluginBuilder/Controllers/AdminController.cs [hunk 3]
```diff
@@ -341,7 +345,11 @@ public async Task<IActionResult> EmailSender(EmailSenderViewModel model)
         {
             return View(model);
         }
-
+        if (!IsValidEmailList(model.To))
+        {
+            TempData[TempDataConstant.WarningMessage] = "Invalid email format in the 'To' field. Please ensure all emails are valid";
+            return RedirectToAction("Plugins");
+        }
```
Comment (rockstardev) [inline PluginBuilder/Controllers/AdminController.cs:352]: If `To` is invalid on post - we should do `ModelState.AddModelError` and provide notice to user, not redirect to `Plugins`

File: PluginBuilder/Controllers/AdminController.cs [hunk 4]
```diff
@@ -351,18 +359,25 @@ public async Task<IActionResult> EmailSender(EmailSenderViewModel model)
...
         {
-            await emailService.SendEmail(model.To, model.Subject, model.Message);
-            TempData[TempDataConstant.SuccessMessage] = $"Test email sent successfully to {model.To}.";
+            var emailsSent = await emailService.SendEmail(model.To, model.Subject, model.Message);
+            TempData[TempDataConstant.SuccessMessage] = $"Emails sent successfully to {emailsSent.Count} recipient(s).";
         }
         catch (Exception ex)
-        {
+        { 
             ModelState.AddModelError(string.Empty, $"Failed to send test email: {ex.Message}");
```
Comment (rockstardev) [inline PluginBuilder/Controllers/AdminController.cs:367]: We don't have Error Summary on view, so this error will not be displayed. Better use TempData Warning message

**Review by rockstardev — CHANGES_REQUESTED** [review]
Just need few changes. It would be good if we could do regex validation client side so users can't enter invalid emails and send them to server side for rejection.

**Comment by TChukwuleta** [comment]
Hey Uncle @rockstardev .. made the changes.. I have also included client side validations for the email

**Review by rockstardev — APPROVED** [review]
Looks great, thanks!

**Pull request merged**